### ISSUE TITLE
Switch remaining jUnit 4 tests to jUnit jupiter

### ISSUE THREAD

**Issue by OlgaMaciaszek** [spring-cloud/spring-cloud-commons#973]

**Comment by spencergibb** [comment]
Assigning to @robertmcnees

**Comment by deepakchethan** [comment]
Hello @spencergibb, Can I take this up if nobody is working on it already? Thanks in advance!

**Comment by spencergibb** [comment]
Someone is assigned already

**Comment by robertmcnees** [comment]
Moved back to open so more modules can be converted.

**Comment by OlgaMaciaszek** [comment]
@ManasviGoyal is working on reducing method and class visibility after the conversion to JUnitJupiter.

**Comment by timtebeek** [comment]
Gave it a quick try to convert more classes over to JUnit 5; outcome here:
https://github.com/spring-cloud/spring-cloud-commons/compare/main...timtebeek:increase-junit-platform-usage
This is mostly the result of applying the [JUnit4 to 5 conversion from OpenRewrite](https://docs.openrewrite.org/tutorials/migrate-from-junit-4-to-junit-5).

I had not yet created a merge request (although willing) as it appeared @ManasviGoyal was also working on this previously, and I don't know the current state of that effort.

I had dropped the Adhoc suites for now, as they are ignored currently, and converting them to disabled suites [would require additional depencies](https://junit.org/junit5/docs/current/user-guide/#junit-platform-suite-engine-example). Please let me know if these should be restored as they were or converted with the added dependencies.

There's quite a few tests left over that rely on the `ModifiedClassPathRunner`; is anyone aware of a similar mechanism for JUnit Jupiter? I'm aware of the [FilteredClassLoader](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#features.developing-auto-configuration.testing.overriding-classpath) when testing Auto-configuration, but I don't know it that would be applicable here.

**Comment by timtebeek** [comment]
Any replacement for the `ModifiedClassPathRunner` could also help complete the Jupiter migration in https://github.com/spring-cloud/spring-cloud-gateway/pull/2387, so mentioning that here to have a feedback loop back to that issue as well.

**Comment by timtebeek** [comment]
Noticed there's a [ModifiedClassPathExtension](https://github.com/spring-projects/spring-boot/blob/main/spring-boot-project/spring-boot-tools/spring-boot-test-support/src/main/java/org/springframework/boot/testsupport/classpath/ModifiedClassPathExtension.java) in Spring Boot; would that be suitable here as well?

**Comment by spencergibb** [comment]
yes, I have local branch with a copy, but there are errors I haven't had time to work thru.
